charmcraft/application/commands/lifecycle.py

```diff
@@ -148,6 +149,68 @@ def get_managed_cmd(self, parsed_args: argparse.Namespace) -> list[str]:
     @override
     def run(self, parsed_args: argparse.Namespace, step_name: str | None = None) -> None:
         """Run a lifecycle step command."""
+        # TODO: When lifecycle commands are replaced with craft-application ExtensibleCommands
```

Comment:
@cmatsuoka would this be better off as a craft-parts prologue? (I would attach it as a method on the `LifecycleService` in that case.)